### PR TITLE
Fix run_tests pip install error

### PR DESCRIPTION
**Pull request by stanley-cheung**
**Problem**

The following commands fail at `master`, if I run them manually and locally:

```
$ ./tools/run_tests/run_tests.py -l c++ --use_docker
$ ./tools/run_tests/run_tests.py -l php7 --use_docker
$ ./tools/run_tests/run_tests.py -l ruby --use_docker
$ ./tools/run_tests/run_tests.py -l grpc-node --use_docker
```

The following is the error:

```
Step 11/16 : RUN pip install --upgrade pip==19.3.1                                                            
 ---> Running in b913525a0fd2                                                                                
/usr/local/lib/python2.7/dist-packages/pkg_resources/py2_warn.py:21: UserWarning: Setuptools will stop working on Python 2                                                                                                
************************************************************                                                  
You are running Setuptools on Python 2, which is no longer                                                    
supported and                                                                                                
>>> SETUPTOOLS WILL STOP WORKING <<<                                                                          
in a subsequent release (no sooner than 2020-04-20).                                                          
Please ensure you are installing                                                                              
Setuptools using pip 9.x or later or pin to `setuptools<45`                                                  
in your environment.                                                                                          
If you have done those things and are still encountering                                                      
this message, please follow up at                                                                            
https://bit.ly/setuptools-py2-warning.                                                                        
************************************************************                                                  
  sys.version_info < (3,) and warnings.warn(pre + "*" * 60 + msg + "*" * 60)                                  
Traceback (most recent call last):                                                                            
  File "/usr/bin/pip", line 9, in <module>                                                                    
    load_entry_point('pip==1.5.6', 'console_scripts', 'pip')()                                                
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 490, in load_entry_point      
    return get_distribution(dist).load_entry_point(group, name)                                              
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2862, in load_entry_point    
    return ep.load()                                                                                          
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2453, in load                
    return self.resolve()                                                                                    
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2459, in resolve              
    module = __import__(self.module_name, fromlist=['__name__'], level=0)                                    
  File "/usr/lib/python2.7/dist-packages/pip/__init__.py", line 74, in <module>                              
    from pip.vcs import git, mercurial, subversion, bazaar  # noqa                                            
  File "/usr/lib/python2.7/dist-packages/pip/vcs/mercurial.py", line 9, in <module>                          
    from pip.download import path_to_url                                                                      
  File "/usr/lib/python2.7/dist-packages/pip/download.py", line 25, in <module>                              
    from requests.compat import IncompleteRead                                                                
ImportError: cannot import name IncompleteRead    
```

**Fix**

1. I noticed that I can move this template `gcp_api_libraries.include` to be just under some other python deps `python_deps.include` so that we don't need to re-install `python-pip`. I made sure that after this PR, every instance of `gcp_api_libraries.include` is now _below_ something that _already_ installed `pip`. There are only 2 variations anyway.
2. [Question to reviewer]: is `easy_install pip` the right way to install `pip`? This way it gets rid of the original error in the bug, but I am not sure if this is the most _correct_ way to install `pip`. [Edit:] Looks like `get-pip.py` should be the way to go
3. [Question to reviewer]: I wonder why this was undetected before? Is it because the tests are doing a `docker pull` of a previously already built docker image from docker hub (the hash of the `Dockerfile` didn't change most likely recently), instead of re-building these images every time from these `Dockerfile` (for which, running the `run_tests.py` script locally will do)?
4. I am running the script `tools/dockerfile/push_testing_images.sh` to build and upload these new images. It's going to take a while.
   - [Edit:] I got into trouble where apparently `easy_install` is not installed despite installing `apt-get install python-setuptools` in Debian Buster. Looks like `easy_install` is no longer being part of the `python-setuptools` package after version 39. (And buster by default installs version 40). So looks like `get-pip.py` is the way to go.
   - Done running `tools/dockerfile/push_testing_images.sh`

### CODE REVIEW

**Comment by jtattermusch** [comment]
> **Problem**
> 
> The following commands fail at `master`:
> 
> ```
> $ ./tools/run_tests/run_tests.py -l c++ --use_docker
> $ ./tools/run_tests/run_tests.py -l php7 --use_docker
> $ ./tools/run_tests/run_tests.py -l ruby --use_docker
> $ ./tools/run_tests/run_tests.py -l grpc-node --use_docker
> ```
> 
> The following is the error:
> 
> ```
> Step 11/16 : RUN pip install --upgrade pip==19.3.1                                                            
>  ---> Running in b913525a0fd2                                                                                
> /usr/local/lib/python2.7/dist-packages/pkg_resources/py2_warn.py:21: UserWarning: Setuptools will stop working on Python 2                                                                                                
> ************************************************************                                                  
> You are running Setuptools on Python 2, which is no longer                                                    
> supported and                                                                                                
> >>> SETUPTOOLS WILL STOP WORKING <<<                                                                          
> in a subsequent release (no sooner than 2020-04-20).                                                          
> Please ensure you are installing                                                                              
> Setuptools using pip 9.x or later or pin to `setuptools<45`                                                  
> in your environment.                                                                                          
> If you have done those things and are still encountering                                                      
> this message, please follow up at                                                                            
> https://bit.ly/setuptools-py2-warning.                                                                        
> ************************************************************                                                  
>   sys.version_info < (3,) and warnings.warn(pre + "*" * 60 + msg + "*" * 60)                                  
> Traceback (most recent call last):                                                                            
>   File "/usr/bin/pip", line 9, in <module>                                                                    
>     load_entry_point('pip==1.5.6', 'console_scripts', 'pip')()                                                
>   File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 490, in load_entry_point      
>     return get_distribution(dist).load_entry_point(group, name)                                              
>   File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2862, in load_entry_point    
>     return ep.load()                                                                                          
>   File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2453, in load                
>     return self.resolve()                                                                                    
>   File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2459, in resolve              
>     module = __import__(self.module_name, fromlist=['__name__'], level=0)                                    
>   File "/usr/lib/python2.7/dist-packages/pip/__init__.py", line 74, in <module>                              
>     from pip.vcs import git, mercurial, subversion, bazaar  # noqa                                            
>   File "/usr/lib/python2.7/dist-packages/pip/vcs/mercurial.py", line 9, in <module>                          
>     from pip.download import path_to_url                                                                      
>   File "/usr/lib/python2.7/dist-packages/pip/download.py", line 25, in <module>                              
>     from requests.compat import IncompleteRead                                                                
> ImportError: cannot import name IncompleteRead    
> ```
> 
> **Fix**
> 
> 1. I noticed that I can move this template `gcp_api_libraries.include` to be just under some other python deps `python_deps.include` so that we don't need to re-install `python-pip`. I made sure that after this PR, every instance of `gcp_api_libraries.include` is now _below_ something that _already_ installed `pip`. There are only 2 variations anyway.
> 2. [Question to reviewer]: is `easy_install pip` the right way to install `pip`? This way it gets rid of the original error in the bug, but I am not sure if this is the most _correct_ way to install `pip`. [Edit:] Looks like `get-pip.py` should be the way to go

I'm surprised that there's not a ready-to-use python pip package.

> 3. [Question to reviewer]: I wonder why this was undetected before? Is it because the tests are doing a `docker pull` of a previously already built docker image from docker hub (the hash of the `Dockerfile` didn't change most likely recently), instead of re-building these images every time from these `Dockerfile` (for which, running the `run_tests.py` script locally will do)?

Yes, exactly. You can trigger the same behavior locally when you set this
https://github.com/grpc/grpc/blob/5a5105b89c1ebedf1a86dd185afcef64a9555f78/tools/internal_ci/helper_scripts/prepare_build_linux_rc#L29
that will cause all the docker to be obtained by `docker pull` instead of trying to build them.
In fact, I was thinking of making that behavior the default, because it's way less fragile (and faster).


> 4. I am running the script `tools/dockerfile/push_testing_images.sh` to build and upload these new images. It's going to take a while.
>    
>    * [Edit:] I got into trouble where apparently `easy_install` is not installed despite installing `apt-get install python-setuptools` in Debian Buster. Looks like `easy_install` is no longer being part of the `python-setuptools` package after version 39. (And buster by default installs version 40). So looks like `get-pip.py` is the way to go.
>    * Done running `tools/dockerfile/push_testing_images.sh`

File: templates/tools/dockerfile/python_deps.include [hunk 1]
```diff
@@ -6,9 +6,10 @@
 RUN apt-get update && apt-get install -y ${'\\'}
     python-all-dev ${'\\'}
     python3-all-dev ${'\\'}
-    python-pip
+    python-setuptools
 
 # Install Python packages from PyPI
+RUN curl https://bootstrap.pypa.io/get-pip.py | python2.7
```
Comment (lidizheng) [inline templates/tools/dockerfile/python_deps.include:12]: 👍

**Comment by stanley-cheung** [comment]
> Using `get-pip.py` script is more robust. Tests are looking good. LGTM.

Thanks for the review!

**Comment by stanley-cheung** [comment]
> Please doublecheck that all the testing images have been uploaded with https://github.com/grpc/grpc/blob/master/tools/dockerfile/push_testing_images.sh.

Yes verified. Thanks!

**Pull request merged**